lib/ansible/modules/database/mongodb/mongodb_replicaset.py

```diff
@@ -0,0 +1,407 @@
+    client["admin"].command('replSetInitiate', conf)
+
+
+def replicaset_remove(module, client, replica_set):
+    raise NotImplementedError
+    # exists = replicaset_find(client, replica_set)
+    # if exists:
+    #    if module.check_mode:
+    #        module.exit_json(changed=True, replica_set=replica_set)
+    #    db = client[db_name]
+    #    db.remove_user(replica_set)
+    # else:
+    #    module.exit_json(changed=False, user=user)
+
+
+def load_mongocnf():
+    config = configparser.RawConfigParser()
+    mongocnf = os.path.expanduser('~/.mongodb.cnf')
+
+    try:
+        config.readfp(open(mongocnf))
+    except (configparser.NoOptionError, IOError):
+        return False
+
+    creds = dict(
+        user=config.get('client', 'user'),
+        password=config.get('client', 'pass')
+    )
+
+    return creds
+
+
+# =========================================
+# Module execution.
+#
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            login_user=dict(type='str'),
+            login_password=dict(type='str', no_log=True),
+            login_database=dict(type='str', default="admin"),
+            login_host=dict(type='str', default="localhost"),
+            login_port=dict(type='int', default=27017),
+            replica_set=dict(type='str', default="rs0"),
+            members=dict(type='list'),
+            arbiter_at_index=dict(type='int'),
+            validate=dict(type='bool', default=True),
+            ssl=dict(type='bool', default=False),
+            ssl_cert_reqs=dict(type='str', default='CERT_REQUIRED', choices=['CERT_NONE', 'CERT_OPTIONAL', 'CERT_REQUIRED']),
+            protocolVersion=dict(type='int', default=1, choices=[0, 1]),
+            chainingAllowed=dict(type='bool', default=True),
+            heartbeatTimeoutSecs=dict(type='int', default=10),
+            electionTimeoutMillis=dict(type='int', default=10000)
+        ),
+        supports_check_mode=True,
+    )
+
+    if not HAS_PYMONGO:
+        module.fail_json(msg='the python pymongo module is required')
+
+    login_user = module.params['login_user']
+    login_password = module.params['login_password']
+    login_database = module.params['login_database']
+    login_host = module.params['login_host']
+    login_port = module.params['login_port']
+    replica_set = module.params['replica_set']
+    members = module.params['members']
+    arbiter_at_index = module.params['arbiter_at_index']
+    validate = module.params['validate']
+    ssl = module.params['ssl']
+    protocolVersion = module.params['protocolVersion']
+    chainingAllowed = module.params['chainingAllowed']
+    heartbeatTimeoutSecs = module.params['heartbeatTimeoutSecs']
+    electionTimeoutMillis = module.params['electionTimeoutMillis']
+
+    if validate:
+        if len(members) <= 2 or len(members) % 2 == 0:
+            raise UserWarning("MongoDB Replicaset validation failed. Invalid number of replicaset members.")
```

Comment:
This should be using `module.fail_json()` instead.

Reply:
Done.